lib/claim_letters/claim_letter_downloader.rb

```diff
@@ -27,16 +27,19 @@ def get_letters
     end
 
     def get_letter(document_id)
-      verify_letter_in_folder(document_id)
+      letter_details = get_letter_details(document_id)
+      raise Common::Exceptions::RecordNotFound, document_id if letter_details.blank?
+
+      filename = filename_with_date('ClaimLetter', letter_details[0][:received_at])
```

Comment:
Nit: `'ClaimLetter'` is a magic string, might be good to place it in a constant in the file somewhere.